tests/test_2678_same_backend.py

```diff
@@ -0,0 +1,19 @@
+# BSD 3-Clause License; see https://github.com/scikit-hep/awkward-1.0/blob/main/LICENSE
+
+import numpy as np  # noqa: F401
+import pytest  # noqa: F401
+
+import awkward as ak
+
+
+def test_where():
+    result = ak.where(
+        [True, False, False], ak.to_backend([1, 2, 3], "typetracer"), [4, 5, 6]
+    )
+    assert ak.backend(result) == "typetracer"
```

Comment:
This just clarifies that the changes in this PR didn't break mixed-backends for `ak.where`.